pkg/components/mysql/mysql.go

```diff
@@ -56,7 +56,24 @@ func Cluster(mattermost *mattermostv1alpha1.ClusterInstallation) *mysqlOperator.
 	if mattermost.Spec.Database.InitBucketURL != "" && mattermost.Spec.Database.BackupRestoreSecretName != "" {
 		mysql.Spec.InitBucketURL = mattermost.Spec.Database.InitBucketURL
 		mysql.Spec.InitBucketSecretName = mattermost.Spec.Database.BackupRestoreSecretName
+	} else if mattermost.Spec.Database.Secret != "" {
+		mysql.Spec.SecretName = mattermost.Spec.Database.Secret
 	}
 
 	return mysql
 }
+
+// Secret returns the secret name created to use together with MySQL deployment
+func Secret(mattermost *mattermostv1alpha1.ClusterInstallation) *corev1.Secret {
```

Comment:
same for the function name, if i look this what this should do?

Reply:
looks like this function is no longer required because of some refactoring on master so removed this.